addons/udes_priorities/models/priority.py

```diff
@@ -84,45 +105,86 @@ def _has_duplicate_names_for_same_pick_type(self):
+                [
+                    ("priority", "=", reference_id),
+                    ("state", "not in", ("cancel", "done")),
+                ]
+            )
 
-        install_mode = self.env.context.get("install_mode", False)
+    def _is_install_mode(self):
+        return self.env.context.get("install_mode", False)
 
-        if install_mode:
+    @api.multi
+    def _validate_picking_types_changes(self, values):
+        if self._is_install_mode():
             return
 
-        priorities_with_outstanding_picks = self.browse()
-        for priority in self:
-            if priority._number_of_oustanding_pickings() > 0:
-                priorities_with_outstanding_picks |= priority
-
-        if priorities_with_outstanding_picks:
-            raise UserError(
-                _("There are outstanding pickings for priorities: {}").format(
-                    ", ".join(priorities_with_outstanding_picks.mapped("name"))
-                )
+        if "picking_type_ids" in values:
+            PickingType = self.env["stock.picking.type"]
+            message = (
+                "Invalid changes, you cannot remove the following transfer types as there are outstanding "
+                "transfers\n "
             )
+            invalid = False
+            old_ids = self.picking_type_ids.ids
+            new_ids = values["picking_type_ids"][0][2]
+
+            for old_pick_type_id in old_ids:
+                if old_pick_type_id not in new_ids:
+                    outstanding_pickings = self._get_outstanding_pickings(
+                        self.reference, pick_type_id=old_pick_type_id
+                    )
+                    if outstanding_pickings:
+                        invalid = True
+                        picking_type = PickingType.search(
+                            [("id", "=", old_pick_type_id)]
+                        )
+                        message += "\n" + picking_type.name
+            if invalid:
+                raise UserError(message)
 
     @api.multi
     def write(self, values):
-        if values and any(k in PROTECTED_FIELDS for k in values):
-            self._check_for_oustanding_pickings()
+        if any(k in MAIN_PROTECTED_FIELDS for k in values):
+            if self._is_install_mode():
+                return
+            raise UserError("Cannot change protected fields.")
+
+        if self._is_trusted_user():
+            self._validate_picking_types_changes(values)
+        else:
+            if any(k in PROTECTED_FIELDS for k in values):
+                raise UserError("You do not have the rights for making changes.")
         return super().write(values)
 
     @api.multi
     def unlink(self):
-        self._check_for_oustanding_pickings()
+        if self._is_trusted_user() is False:
+            raise UserError("You do not have the rights for making changes.")
+        for priority in self:
+            if priority.outstanding_picking_count > 0:
```

Comment:
Dont need > 0 i think should be enough:
`if priority.outstanding_picking_count:`